crawler/processing/rabbit_message_processor.py

```diff
@@ -1,6 +1,7 @@
 import logging
 from typing import Any, Dict, Optional, cast
 
+import crawler.helpers.general_helpers as general_helpers
```

Comment:
Why in this case it is importing the module instead of the functions from the module we will use?

Reply:
I probably should change this to match other imports of internal code.  I'm not sure if we do have a standard in our code base since we tend to import whole built-in modules, like `import logging` above.

The reason I did it here was because it makes the patch in testing a bit more logical and I knew I needed to patch these.

If you use `from x.y.z import method` then you have to patch `crawler.processing.rabbit_message_processor.method` because it becomes part of the module it was imported into.

If you use `import x.y.z` then you can patch `x.y.z.method` which feels more logical. 

We probably shouldn't write the code to make unit tests more logical though.  Unit tests are a second class citizen when considering the design of the code.

Reply:
It could make more sense on unit test mocking, but I also think that could originate more problems than it solves. Most of what people learn on programming is from code of others and if we write code that works in the logical way, a new programmer could get the impression that mocking with pytest works differently to how it works in reality.

Reply:
Neither of these approaches is wrong.  It's a stylistic choice.  Pytest does work in two different ways depending on how you decided to do your import.  It's because of the way Python works.  If anything, you probably do expect to be able to patch `x.y.z.method` but when you do, your mock isn't used in the tests.  In almost all our unit tests, you have to mock your dependencies as if they are defined in the class you're doing tests on.  This is because we use the `from x.y.z import method` approach for all our imports of Crawler code.  I will change this one to be consistent, but the way we have to do testing is confusing to someone unfamiliar with how and when Python imports work.

Reply:
You can see how the tests are affected by this change in this commit:

https://github.com/sanger/crawler/commit/5f23519f30c8b449e5711932ebc639e72af36b27